wagtail/wagtailcore/models.py

```diff
@@ -391,6 +391,15 @@ def _get_autogenerated_slug(self, base_slug):
         candidate_slug = base_slug
         suffix = 1
         parent_page = self.get_parent()
+        slug_parts = base_slug.split('-')
+
+        try:
+            slug_parts[-1] = str(int(slug_parts[-1]) + 1)
+            candidate_slug = '-'.join(slug_parts)
+        except ValueError:
+            # If the last part of the slug fails to be converted
+            # to an int then simply move on to auto incrementing
+            pass
 
         while not Page._slug_is_available(candidate_slug, parent_page, self):
             # try with incrementing suffix until we find a slug which is available
```

Comment:
I think that would always be the desired behaviour for numbers that the user has explicitly entered as part of the slug - if there's an existing page called 'the-magnificent-7', a new page with the same slug should be turned into 'the-magnificent-7-2', not 'the-magnificent-8' :-) (On the other hand, if the slug they entered was 'the-magnificent', then it *would* go through 'the-magnificent-2', 'the-magnificent-3' etc...)

We do of course want to avoid the situation where a slug that's been given an automatic number suffix gets another one added later on - but I can't see any scenario where this would happen.